rxandroid/src/main/java/rx/android/widget/SearchViewQueryText.java

```diff
@@ -0,0 +1,143 @@
+
+                    @Override
+                    public boolean onQueryTextChange(String newText) {
+                        return false;
+                    }
+                };
+
+                composite.addOnQueryTextListener(listener);
+                subscriber.add(AndroidSubscriptions.unsubscribeInUiThread(new Action0() {
+                    @Override
+                    public void call() {
+                        composite.removeOnQueryTextListener(listener);
+                    }
+                }));
+            }
+        });
+    }
+
+    public static Observable<String> change(final SearchView searchView) {
+        return Observable.create(new Observable.OnSubscribe<String>() {
+            @Override
+            public void call(final Subscriber<? super String> subscriber) {
+                Assertions.assertUiThread();
+                final CompositeOnQueryTextSubmitListener composite = CachedListeners.getFromViewOrCreate(searchView);
+                final SearchView.OnQueryTextListener listener = new SearchView.OnQueryTextListener() {
+                    @Override
+                    public boolean onQueryTextSubmit(String query) {
+                        return false;
+                    }
+
+                    @Override
+                    public boolean onQueryTextChange(String newText) {
+                        subscriber.onNext(newText);
+                        return false;
+                    }
+                };
+
+                composite.addOnQueryTextListener(listener);
+                subscriber.add(AndroidSubscriptions.unsubscribeInUiThread(new Action0() {
+                    @Override
+                    public void call() {
+                        composite.removeOnQueryTextListener(listener);
+                    }
+                }));
+            }
+        });
+    }
+
+
+
+    private static class CompositeOnQueryTextSubmitListener implements SearchView.OnQueryTextListener {
+
+        private final List<SearchView.OnQueryTextListener> listeners = new ArrayList<SearchView.OnQueryTextListener>();
+
+        public boolean addOnQueryTextListener(final SearchView.OnQueryTextListener listener) {
+            return listeners.add(listener);
+        }
+
+        public boolean removeOnQueryTextListener(final SearchView.OnQueryTextListener listener) {
+            return listeners.remove(listener);
+        }
+
+        @Override
+        public boolean onQueryTextSubmit(String query) {
+            for (SearchView.OnQueryTextListener listener: listeners) {
+                listener.onQueryTextSubmit(query);
+            }
+            return false;
+        }
+
+        @Override
+        public boolean onQueryTextChange(String newText) {
+            for (SearchView.OnQueryTextListener listener: listeners) {
+                listener.onQueryTextChange(newText);
+            }
+            return false;
+        }
+
+    }
+
+
+    private static class CachedListeners {
```

Comment:
We have this pattern enough that I wonder if it would be prudent to consolidate things a bit if possible.